File: backend-leocloud-beeyond/src/main/java/at/htl/beeyond/bean/TemplateBean.java
package at.htl.beeyond.bean;

import at.htl.beeyond.entity.Template;
import com.fasterxml.jackson.databind.ObjectMapper;
import io.quarkus.runtime.StartupEvent;

import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.event.Observes;
import javax.transaction.Transactional;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.stream.Collectors;

@ApplicationScoped
class TemplateBean {
    @Transactional
    void init(@Observes StartupEvent event) throws IOException {
        var filesIn = getClass().getResourceAsStream("/templates/file-list.txt");
        if (filesIn != null) {
            var files = new BufferedReader(new InputStreamReader(filesIn)).lines().toList();
            for (String filename : files) {
                try (InputStream in = getClass()
                        .getResourceAsStream("/templates/json/" + filename)) {
                    assert in != null;

                    BufferedReader reader = new BufferedReader(new InputStreamReader(in));
                    var jsonString = reader.lines().collect(Collectors.joining("\n"));
                    var newTemplate = new ObjectMapper().readValue(jsonString, Template.class);

                    InputStream contentIn = getClass()
                            .getResourceAsStream("/templates/yml/" + newTemplate.getContent());
                    assert contentIn != null;
                    BufferedReader contentReader = new BufferedReader(new InputStreamReader(contentIn));
                    newTemplate.setContent(
                            contentReader.lines().collect(Collectors.joining("\n"))
                    );
                    contentIn.close();
                    newTemplate.getFields().forEach(fieldName -> fieldName.setTemplate(newTemplate));
                    newTemplate.persist();
                }
            }
        }
    }
}

File: backend-leocloud-beeyond/src/main/java/at/htl/beeyond/bean/TemplateBean.kt
package at.htl.beeyond.bean

import at.htl.beeyond.entity.Template
import com.fasterxml.jackson.databind.ObjectMapper
import io.quarkus.runtime.StartupEvent
import javax.enterprise.context.ApplicationScoped
import javax.enterprise.event.Observes
import javax.transaction.Transactional

@ApplicationScoped
class TemplateBean {
    @Transactional
    fun init(@Observes event: StartupEvent) {
        val filesIn = javaClass.getResourceAsStream("/templates/file-list.txt")
        filesIn?.let { stream ->
            val files = stream.bufferedReader().use { it.readLines() }
            files.forEach { filename ->
                javaClass.getResourceAsStream("/templates/json/$filename")?.let { jsonStream ->
                    val jsonString = jsonStream.bufferedReader().use { it.readText() }
                    val newTemplate = ObjectMapper().readValue(jsonString, Template::class.java)
                    javaClass.getResourceAsStream("/templates/yml/${newTemplate.content}")?.let { contentStream ->
                        val content = contentStream.bufferedReader().use { it.readText() }
                        newTemplate.content = content
                        contentStream.close()
                        newTemplate.fields.forEach { field ->
                            field.template = newTemplate
                        }
                    }
                    javaClass.getResourceAsStream("/templates/svg/${newTemplate.img}")?.let { contentStream ->
                        val content = contentStream.bufferedReader().use { it.readText() }
                        newTemplate.img = content
                        contentStream.close()
                    }
                    newTemplate.persist()
                    jsonStream.close()
                }
            }
            stream.close()
        }
    }
}
